### ISSUE TITLE
How does caffeine match process names?

### ISSUE THREAD

**Issue by protist** [caffeine-ng/caffeine-ng#18]
This is probably related to issue #7. I'm trying to manually activate a running instance of caffeine. I'm not entirely sure how `-a` and `-d` work. I assumed they would trigger an existing instance of caffeine, but I guess not? I just get `CRITICAL:caffeine.main:Caffeine is already running`.

Hence, I tried to whitelist a custom process instead. I wrote a script that just stays open. In `~/prevent_suspend_process`,
```
#!/usr/bin/env bash
while [[ $response != 'q' ]]; do read -n 1 -p '[q]uit?' response; echo; done
```
However, after adding `prevent_suspend_process` to caffeine's whitelist, it still doesn't trigger. I wonder if the process name is the problem, and it's matching poorly. Is there a way to implement this workaround, so something similar?
```
$ pgrep prevent -a
$ pgrep prevent -af
24169 bash /home/protist/.local/bin/prevent_suspend_process
$ ps aux | grep '[p]revent'
protist  24169  0.0  0.0   9964  2784 pts/0    S+   12:09   0:00 bash /home/protist/.local/bin/prevent_suspend_process
```

**Comment by WhyNotHugo** [comment]
Whitelisting is done by process name, not arguments. The process you're running there is `bash`, with your script as an argument.

You probably want something that generate a binary for this specific usage (though fixing #7 should be a priority, actually).

**Comment by protist** [comment]
Thanks for the response. I couldn't get it to work, so I'll just wait until #7 is fixed.

For the record, I tried a few things to manipulate process names. Starting the process with `exec -a prevent_suspend_process prevent_suspend_process` didn't work. I tried `zsh` and `jobs -Z prevent_suspend_process`, which seemed to work in the reporting, but not in the vanilla `pgrep` search (nor would it trigger caffeine).

i.e. the following worked
```
$ pgrep prevent -af
24169 bash /home/protist/.local/bin/prevent_suspend_process
$ ps aux | grep '[p]revent'
protist  24169  0.0  0.0   9964  2784 pts/0    S+   12:09   0:00 bash /home/protist/.local/bin/prevent_suspend_process
```
But not vanilla `pgrep`
```
$ pgrep prevent -a
```

But oddly
```
$ pgrep zsh -a
...
28210 prevent_suspend_process                         
```

**Comment by protist** [comment]
I had to revisit this recently, and did some more testing. I found a hacky solution that works and might be useful to others.

If I copy a binary and rename it, caffeine will recognise the new name. For example, I copied `sleep` and `yad` to `sleep2` and `yad2`, and put these two strings on caffeine's "automatic activation" list. This worked perfectly, and also without caffeine being activated for the vanilla `sleep` and `yad`.
